frontend/src/metabase/query_builder/selectors.js

```diff
@@ -952,3 +959,15 @@ export const getDataReferenceStack = createSelector(
       ? [{ type: "database", item: { id: dbId } }]
       : [],
 );
+
+export const getNativeQueryFn = createSelector(
```

Comment:
Tried to do it similar to autocomplete function selectors in the same file. So far seems to be the most straightforward and robust approach.